workflow/manager.py

```diff
@@ -6,24 +9,42 @@
 from workflow import query as q, rai
 
 
+@dataclasses.dataclass
+class EngineMetaInfo:
+    engine: str
+    size: str
+    is_default: bool
+
+
 class ResourceManager:
-    logger: logging.Logger
-    rai_config: RaiConfig
+    __logger: logging.Logger
+    __rai_config: RaiConfig
+    __engines: dict[str, EngineMetaInfo]
 
     def __init__(self, logger: logging.Logger, rai_config: RaiConfig):
-        self.logger = logger
-        self.rai_config = rai_config
+        self.__logger = logger
+        self.__rai_config = rai_config
+        self.__engines = {}
```

Comment:
would at any point here be more than two engines (default and a worker needed for a step which has other engine spec) active?

Reply:
not now. But it might be requested to keep engine after step execution. Then it won't require changes in manager logic.